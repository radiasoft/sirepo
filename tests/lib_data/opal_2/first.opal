
OPTION, PSDUMPFREQ=5000;   // 6d data written on steps (h5).
OPTION, STATDUMPFREQ=1;  // Beam Stats written on steps (stat).
OPTION, BOUNDPDESTROYFQ=10; // Delete lost particles, if any       
OPTION, AUTOPHASE= 6; 
OPTION, VERSION = 20000;

Title, string="conversion";

//----------------------------------------------------------------------------
//Global Parameters
//----------------------------------------------------------------------------

REAL n_particles=5E4;

REAL gamma=273.97316778927893;  // 140 MeV electrons

REAL beta=sqrt(1-(1/gamma^2));

REAL P0=gamma*beta*EMASS;    //inital z momentum

value , {gamma*beta, EMASS, P0};


//----------------------------------------------------------------------------
//Bookkeeping
//----------------------------------------------------------------------------

// emit = 0.2e-6
// alpha = 0
// beta = 0.1
// rms emittance

REAL emit_n = 7.8983331046e-07;
REAL emit = emit_n / (beta * gamma);
REAL beta_twi = 2.83302878621;

//----------------------------------------------------------------------------
// Import Lattice definitions
//----------------------------------------------------------------------------

// CALL,FILE="spectrometer.txt";
REAL edge_drift_und2quads = 0;
drift_und2quads: DRIFT, L = 1.0, ELEMEDGE = edge_drift_und2quads;
REAL edge_q_def = 1.0;
q_def: QUADRUPOLE, L = 0.074, K1 = 21.808275280025114, ELEMEDGE = edge_q_def;
REAL edge_drift_quad2quad = 1.074;
drift_quad2quad: DRIFT, L = 0.03, ELEMEDGE = edge_drift_quad2quad;
REAL edge_q_foc = 1.104;
q_foc: QUADRUPOLE, L = 0.074, K1 = -21.808275280025114, ELEMEDGE = edge_q_foc;
REAL edge_drift_quad2dip = 1.1780000000000002;
drift_quad2dip: DRIFT, L = 1.5, ELEMEDGE = edge_drift_quad2dip;
REAL edge_spectr_dipole = 2.678;
spectr_dipole: SBEND, L = 0.49430796473268457, ANGLE = 0.5235987755982988, E1 = 0.0, E2 = 0.0,
        GAP = 0.0, PSI = 0.0, ELEMEDGE = edge_spectr_dipole, DESIGNENERGY = 139.489, FMAPFN = "hard_edge_profile.txt";
REAL edge_drift_dip2dump = 3.1723079647326844;
drift_dip2dump: DRIFT, L = 0.3, ELEMEDGE = edge_drift_dip2dump;

beamline_1: LINE=(drift_und2quads, q_def, drift_quad2quad, q_foc, drift_quad2dip, 
spectr_dipole, drift_dip2dump); 

//----------------------------------------------------------------------------
// INITIAL DISTRIBUTION
//----------------------------------------------------------------------------

// Cold Gaussian distribution for testing purposes
start_gaussian: DISTRIBUTION, TYPE = GAUSS,
        SIGMAX = sqrt(emit * beta_twi ),
        SIGMAPX = sqrt(emit / beta_twi) * beta * gamma,
        SIGMAY = sqrt(emit * beta_twi ),
        SIGMAPY = sqrt(emit / beta_twi ) * beta * gamma,
        SIGMAZ = 0.00065,
        CUTOFFLONG = 0.1, 
        SIGMAPZ = 6.576e-3 * beta * gamma ,   
        EMITTED = False,
        INPUTMOUNITS = None,      
        WRITETOFILE = True; 
 
//----------------------------------------------------------------------------
// Define Field solvers
//----------------------------------------------------------------------------

FS_SC: Fieldsolver, FSTYPE = NONE, // None or FFT
            MX = 32, MY = 32, MT = 32,
            PARFFTX = false, 
            PARFFTY = false, 
            PARFFTT = true,  //parallel in the z direction only
            BCFFTX = open, 
            BCFFTY = open, 
            BCFFTT = open,
            BBOXINCR = 1, 
            GREENSF = INTEGRATED;

//----------------------------------------------------------------------------
// Electron Beam Definition

BEAM1:  BEAM, PARTICLE = ELECTRON, pc = P0, NPART = n_particles,
        BFREQ = 1.0e6, BCURRENT = 0.0, CHARGE = -1;

//----------------------------------------------------------------------------
// Initialize Simulation
//----------------------------------------------------------------------------

TRACK, LINE =beamline_1, BEAM = BEAM1, MAXSTEPS = 10000000, 
    DT = 1.8076756186965821e-12, ZSTOP=3.5;

RUN, METHOD = "PARALLEL-T", BEAM = BEAM1, 
    FIELDSOLVER = FS_SC, DISTRIBUTION = start_gaussian;
ENDTRACK;

Stop;
Quit;
