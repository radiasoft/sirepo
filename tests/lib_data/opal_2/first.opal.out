
Title, string="conversion";

REAL beta = 0.9999933387552905;
REAL beta_twi = 2.83302878621;
REAL edge_drift_dip2dump = 3.1723079647326844;
REAL edge_drift_quad2dip = 1.1780000000000002;
REAL edge_drift_quad2quad = 1.074;
REAL edge_drift_und2quads = 0.0;
REAL edge_q_def = 1.0;
REAL edge_q_foc = 1.104;
REAL edge_spectr_dipole = 2.678;
REAL emit = 2.8829048411612934e-09;
REAL emit_n = 7.8983331046e-07;
REAL gamma = 273.97316778927893;
REAL n_particles = 50000.0;
REAL p0 = 0.13999906027508863;


OPT1: option,psdumpfreq=5000.0,statdumpfreq=1.0,version=20000.0;

DRIFT_DIP2DUMP: DRIFT,l=0.3;
DRIFT_QUAD2DIP: DRIFT,l=1.5;
DRIFT_QUAD2QUAD: DRIFT,l=0.03;
DRIFT_UND2QUADS: DRIFT,l=1.0;
Q_DEF: QUADRUPOLE,k1=21.808275280025114,l=0.074;
Q_FOC: QUADRUPOLE,k1=-21.808275280025114,l=0.074;
SPECTR_DIPOLE: SBEND,angle=0.5235987755982988,designenergy=139.489,fmapfn="hard_edge_profile.txt",l=0.49430796473268457;

"drift_und2quads#0": drift_und2quads,elemedge=0;
"q_def#0": q_def,elemedge=1.0;
"drift_quad2quad#0": drift_quad2quad,elemedge=1.074;
"q_foc#0": q_foc,elemedge=1.104;
"drift_quad2dip#0": drift_quad2dip,elemedge=1.1780000000000002;
"spectr_dipole#0": spectr_dipole,elemedge=2.678;
"drift_dip2dump#0": drift_dip2dump,elemedge=3.1666159294626843;
beamline_1: LINE=("drift_und2quads#0","q_def#0","drift_quad2quad#0","q_foc#0","drift_quad2dip#0","spectr_dipole#0","drift_dip2dump#0");


START_GAUSSIAN: distribution,cutofflong=0.1,inputmounits="NONE",sigmapx=0.008739665247723578,sigmapy=0.008739665247723578,sigmapz=1.8016355501670784,sigmax=9.037340539624537e-05,sigmay=9.037340539624537e-05,sigmaz=0.00065,type="GAUSS",writetofile=true;
FS_SC: fieldsolver,bboxincr=1.0,fstype="NONE",mt=32.0,mx=32.0,my=32.0;
BEAM1: beam,bfreq=1000000.0,charge=-1.0,npart=50000.0,particle="ELECTRON",pc=0.13999906027508863;
TR1: track,beam=BEAM1,dt=1.8076756186965821e-12,line=beamline_1,maxsteps=10000000.0,zstop=3.5;
 run, beam=BEAM1,distribution=start_gaussian,fieldsolver=FS_SC,method="PARALLEL-T";
endtrack;
